src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -436,12 +436,7 @@ private function addValidationSection(ArrayNodeDefinition $rootNode)
                     ->info('validation configuration')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('cache')
-                            ->beforeNormalization()
-                                // Can be removed in 3.0, once ApcCache support is dropped
-                                ->ifString()->then(function ($v) { return 'apc' === $v ? 'validator.mapping.cache.apc' : $v; })
```

Comment:
In order to upgrade from symfony 2.7 to 3.0, he'll need to update to 2.8 first. Then adding the deprecation in 2.8 in a bugfix is the right thing to do, and no one should expect having this service anymore in 3.0.

I removed the BC flag from the PR description accordingly. The existence of the `validator.mapping.cache.apc` service and the removal of the `validator.mapping.cache.doctrine.apc` one in 3.0 (as well as its deprecation in 2.8 instead of first one) is a bug and should be fixed in both branches.